apps/mobile/src/features/settings/choose-avatar/avatars.tsx

```diff
@@ -0,0 +1,29 @@
+import { iconNames } from '@/components/avatar-icon';
+
+import { Box } from '@leather.io/ui/native';
+
+import { AvatarButton } from './avatar-button';
+
+interface AvatarProps {
+  setNewIcon(icon: string): void;
+  currentIcon: string;
+}
+
+export function Avatars({ setNewIcon, currentIcon }: AvatarProps) {
+  return (
+    <Box flexDirection="row" gap="5" flexWrap="wrap">
+      {Object.values(iconNames).map(icon => {
+        return (
+          <AvatarButton
+            isSelected={currentIcon === icon}
+            onPress={() => {
+              setNewIcon(icon);
+            }}
+            key={icon}
+            icon={icon}
+          />
+        );
+      })}
+    </Box>
+  );
+}
```

Comment:
Just style jamming here, but one approach to avoid the event drilling up of the `onPress` and `setNetIcon` could be a render prop (even if these seem out of fashion)

```tsx
interface AvatarListProps {
  currentIcon: string;
  renderAvatar(name): ReactNode;
}
export function AvatarList({ renderAvatar, currentIcon }: AvatarProps) {
  return (
    <Box flexDirection="row" gap="5" flexWrap="wrap">
      {Object.values(iconNames).map(icon => renderAvatar(icon))}
    </Box>
  );
}

// in other component
<Avatars 
  currentIcon={newIcon ?? account.icon} 
  renderAvatar={icon => (
    <AvatarButton
      isSelected={currentIcon === icon}
      onPress={() => {
        setNewIcon(icon);
      }}
      key={icon}
      icon={icon}
    />
  )} 
/>
```

Reply:
or just use `children` and do the map in parent too, so the AvatarList is only just the styles around it